src/lib/storage/table.cpp

```diff
@@ -79,6 +90,35 @@ void Table::_add_value_segment_to_chunk(std::shared_ptr<Chunk>& chunk, const std
   });
 }
 
-void Table::compress_chunk(ChunkID chunk_id) { throw std::runtime_error("Implement Table::compress_chunk"); }
+void Table::_add_dictionary_segment_to_chunk(std::shared_ptr<Chunk>& chunk, const std::string& type,
+                                             const std::shared_ptr<BaseSegment>& value_segment) {
+  resolve_data_type(type, [&](const auto data_type_t) {
+    using ColumnDataType = typename decltype(data_type_t)::type;
+    const auto dictionary_segment = std::make_shared<DictionarySegment<ColumnDataType>>(value_segment);
+    chunk->add_segment(dictionary_segment);
```

Comment:
Because _add_dictionary_segment_to_chunk is called parallel by many threads, this may change the order of the segments in the resulting chunk. We implemented it so that the order does not change but we are not sure if this is a problem.